onnxruntime_extensions/tools/pre_post_processing/steps/vision.py

```diff
@@ -760,3 +760,221 @@ def _create_graph_for_step(self, graph: onnx.GraphProto, onnx_opset: int):
+
+
+    def _create_graph_for_step(self, graph: onnx.GraphProto, onnx_opset: int):
+        input0_type_str, input0_shape_str = self._get_input_type_and_shape_strs(graph, 0)
+        input1_type_str, input1_shape_str = self._get_input_type_and_shape_strs(graph, 1)
+
+        input0_shape_list = input0_shape_str.split(',')
+        assert len(input0_shape_list) == 2, " expected [num_boxes, 4]"
+
+        target_shape_str = f"_{self._step_num}_nms_boxes, 6"
+
+        reduce_score = '(score_select_nm,i64_neg1)' if onnx_opset >= 18 else '<axes=[-1]>(score_select_nm)'
+
+        converter_graph = onnx.parser.parse_graph(
+            f"""\
+            NMS (float[{input0_shape_str}] {self.input_names[0]},float[{input1_shape_str}] {self.input_names[1]}) 
+                => (float[{target_shape_str}] {self.output_names[0]})  
+            {{
+                i64_2 = Constant <value = int64[1] {{2}}>()
+                i64_0 = Constant <value = int64[1] {{0}}>()
+                i64_1 = Constant <value = int64[1] {{1}}>()
+                i64_max_obj = Constant <value = int64[1] {{{self.max_detections_}}}>()
+                i64_neg1 = Constant <value = int64[1] {{-1}}>()
+                fp32_iou_th = Constant <value = float[1] {{{self.iou_threshold_}}}>()
+                fp32_score_th = Constant <value = float[1] {{{self.score_threshold_}}}>()
+
+                boxes_i = Identity ({self.input_names[0]})
+                scores_i = Identity({self.input_names[1]})
+                scores_c_b = Transpose<perm=[1,0]>(scores_i)
+                batch_boxes = Unsqueeze(boxes_i, i64_0)
+                batch_scores = Unsqueeze(scores_c_b, i64_0)
+
+                nmsbox = NonMaxSuppression<center_point_box =1>(batch_boxes, batch_scores, i64_max_obj,fp32_iou_th,fp32_score_th)
+                classes_i64 = Gather <axis=-1>(nmsbox,i64_1)
+                class_select = Cast <to = 1>(classes_i64)
+
+                boxes_idx_us = Gather <axis=-1>(nmsbox,i64_2)
+                boxes_idx = Squeeze(boxes_idx_us, i64_neg1)
+                boxes_select = Gather <axis=0>(boxes_i, boxes_idx)
+
+                score_select_nm = Gather <axis=0>(scores_i, boxes_idx)
+                score_select = ReduceMax{reduce_score}
+
+                {self.output_names[0]} = Concat <axis = -1> (boxes_select, score_select, class_select)
+            }}
+            """
+        )
+        return converter_graph
+
+
+class LinearMapBox(Step):
+    """
+    Mapping boxes coordinate to scale in original image.
+    The coordinate of boxes from detection model is relative to the input image of network, 
+    image is scaled and padded/cropped. So we need to do a linear mapping to get the real coordinate of original image.
+    input:
+        ori_img: original image decoded from jpg/png<uint8_t>[H, W, 3<BGR>]
+        scaled_img: scaled image, but without padding/crop[<uint8_t>[H1, W1, 3<BGR>]
+        net_in_img: scaled image and with padding/crop[<uint8_t>[H2, W3, 3<BGR>]
```

Comment:
We have to do `OFFSET` as Lettbox Step padded/cropped the resized image. It means that coordinates are centered on cropped/padded image.
The point with coordinates (0, 0) should be based on resized image instead of cropped image.